packages/ds-component/src/index.ts

```diff
@@ -333,13 +334,28 @@ const transformDSInteraction = (message: Message): InteractionsById => {
   );
 };
 
+/**
+ * Transform for date ranges
+ */
+const toDateRanges = (message: Message): DateRange => {
```

Comment:
One small thing. Instead of this always returning an array, I think it should return undefined when date ranges aren't passed in the message from dataStudio. We can just change 341 to be
```
if (message.dataResponse.dateRanges === undefined) {
  return undefined
}
const dateRanges = message.dataResponse.dateRanges!;
```